src/main/java/org/hydev/mcpm/client/database/SuperLocalPluginTracker.java

```diff
@@ -0,0 +1,582 @@
+        for (PluginYml plugin : installedPlugins) {
+            try {
+                if (compareVersion(plugin.name(), searchPackagesBoundary)) {
+                    outdatedPlugins.add(plugin);
+                }
+            } catch (Exception e) {
+                System.out.println("Error checking if plugin " + plugin.name() + " is outdated");
+            }
+        }
+
+        return outdatedPlugins;
+    }
+
+    /**
+     * Compare whether the locally installed version of the plugin matches the
+     * version on the server.
+     * If yes, return true. If no, return false.
+     *
+     * @return True if the local version of plugin with name name is outdated, false
+     *         otherwise
+     */
+    public Boolean compareVersion(String name, SearchPackagesBoundary searchPackagesBoundary) {
+        try {
+            File pluginYmlPath = getPluginFile(name);
+            PluginYml currPlugin = readMeta(pluginYmlPath);
+            String localVersion = currPlugin.version();
+
+            SearchPackagesInput searchPackagesInput = new SearchPackagesInput(SearchPackagesType.BY_NAME, name,
+                    false);
+            SearchPackagesResult searchPackagesResult = searchPackagesBoundary.search(searchPackagesInput);
+
+            // Get the version of the plugin from the server: Query for all, see if there's
+            // a match
+            if (searchPackagesResult.state().equals(SearchPackagesResult.State.SUCCESS)) {
+                for (PluginModel plugin : searchPackagesResult.plugins()) {
+                    PluginVersion latest = plugin.getLatestPluginVersion().orElse(null);
+                    if (latest != null) {
+                        if (latest.meta().version().equals(localVersion)) {
+                            return true;
+                        }
+                    }
+
+                }
+            } else {
+                System.out.println("Error getting hash from server");
+                return false;
+            }
+
+        } catch (Exception e) {
+            System.out.println("Error fetching local plugin version");
+            return false;
+        }
+        return false;
+    }
+
+    /**
+     * Get whether a local plugin File matches the version of the plugin on the
+     * server
+     *
+     * @param local  the local plugin File
+     * @param remote a PluginModel object representing the plugin on the server
+     * @return List of plugin names
+     */
+    public Boolean compareVersionNew(File local, PluginModel remote) {
+        try {
+            PluginYml localPlugin = readMeta(local);
+            String localVersion = localPlugin.version();
+            String remoteVersion = remote.versions().get(0).meta().version();
+            return localVersion.equals(remoteVersion);
+        } catch (Exception e) {
+            System.out.println("Error comparing versions");
+            return false;
+        }
```

Comment:
You shouldn't catch all exceptions, and probably shouldn't print anywhere in the class because reporting errors to the user should be handled by the controller, and logging is actually handled differently in different environments. (e.g. if you use stderr, a Minecraft user executing the command wouldn't see the error)